ComponentTextKit/TextKit/CKTextKitRenderer.mm

```diff
@@ -125,10 +125,16 @@ - (NSUInteger)lineCount
 {
   __block NSUInteger lineCount = 0;
   [_context performBlockWithLockedTextKitComponents:^(NSLayoutManager *layoutManager, NSTextStorage *textStorage, NSTextContainer *textContainer) {
-    for (NSRange lineRange = { 0, 0 }; NSMaxRange(lineRange) < [layoutManager numberOfGlyphs]; lineCount++) {
-      [layoutManager lineFragmentRectForGlyphAtIndex:NSMaxRange(lineRange) effectiveRange:&lineRange];
+    NSUInteger numberOfGlyphs = [layoutManager numberOfGlyphs];
+    for (NSRange lineRange = { 0, 0 }; NSMaxRange(lineRange) < numberOfGlyphs;) {
+      CGRect lineRect = [layoutManager lineFragmentRectForGlyphAtIndex:NSMaxRange(lineRange) effectiveRange:&lineRange];
+      if (CGRectIsEmpty(lineRect)) {
```

Comment:
At least that is how tests pass, documentation doesn't specify but `CGRectIsEmpty` handles a null rectangle or a valid rectangle with zero height or width.